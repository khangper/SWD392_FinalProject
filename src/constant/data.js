export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
