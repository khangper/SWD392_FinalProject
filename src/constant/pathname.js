export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
