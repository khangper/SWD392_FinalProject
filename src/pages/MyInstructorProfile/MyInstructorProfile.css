/* SUBSCRIPTIONS */
.mip-subscription {
  display: flex;
  margin: 10px;
}

.mip-instructor-profile {
  margin-top: 15px;
  margin-bottom: 25px;
  background: #fff;
  width: 100%;
  float: left;
  padding: 10px;
  border-radius: 3px;
  border: 1px solid #efefef;
  transition: all 0.2s ease-in-out;
  padding-right: 10px;
}

.mip-subcribe {
  display: inline-block;
  padding: 10px 20px;
  border: none;
  border-radius: 5px;
  background: #f00;
  color: #fff;
  cursor: pointer;
  transition: background 0.3s;
}

.mip-subcribe:hover {
  background-color: #333;
}

.mip-notification {
  display: inline-block;
  padding: 10px 20px;
  border-radius: 5px;
  cursor: pointer;
  transition: background 0.3s;
  background: #dad6d6;
  margin-left: 10px;
  border: 1px solid #fff;
}

.mip-notification .cart-icon {
  width: 12px;
}

.mip-oiv-course {
  background: #fff;
  border: 1px solid #e5e5e5;
  border-radius: 8px;
  overflow: hidden;
  width: 100%;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
}

.mip-oiv-course-detail {
  display: flex;
}

.mip-cart {
  display: none;
  border: none;
  padding-left: 670px;
  background: white;
}

.mip-oiv-course-detail:hover .mip-cart {
  display: block;
}

.subscribe-btn {
  background: #F9BF4B !important;
  color: #fff;
}

.subscribe-btn:hover {
  background: #e3a62b !important;
}

.other_instructor_view-nav .tab.active {
  border-bottom: 4px solid #196c7e !important;
}

.tab.active::after{
  background: none !important;
}

.oiv-cmnt-btn{
  background: #196c7e !important;
}

.oiv-badge_seller{
  background-color: #02dd5a !important;
}

.crse_reviews{
  background-color: #e3a62b;
}