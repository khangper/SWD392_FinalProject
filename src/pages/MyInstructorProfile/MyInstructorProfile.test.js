import React from "react";
import { renderWithProviders } from "../../constant/test-util";
import MyInstructorProfile from "./MyInstructorProfile";
import { screen } from "@testing-library/react";

describe("MyInstructorProfile", () => {
    beforeAll(() => {
        jest.spyOn(console, "error").mockImplementation(() => { });
        jest.spyOn(console, "log").mockImplementation(() => { });
    });

    afterAll(() => {
        console.error.mockRestore();
        console.log.mockRestore();
    });
    it("renders the MyInstructorProfile component", async () => {
        const { container } = renderWithProviders(<MyInstructorProfile />);
        expect(container).toBeDefined();
        expect(container).toMatchSnapshot();
    });
})