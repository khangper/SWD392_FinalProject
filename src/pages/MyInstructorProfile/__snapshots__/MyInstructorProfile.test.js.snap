// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyInstructorProfile renders the MyInstructorProfile component 1`] = `
<div>
  <div>
    <div
      class="other_instructor_view-wrapper"
    >
      <div
        class="card"
      >
        <div
          class="other_instructor_view-container"
        >
          <div
            class="other_instructor_view-row"
          >
            <div
              class="other_instructor_view-col-md-12"
            >
              <div
                class="other_instructor_view-section"
              >
                <div
                  class="other_instructor_view-row2"
                >
                  <div
                    class="other_instructor_view-col-lg-6"
                  >
                    <div
                      class="other_instructor_view-avatar"
                    >
                      <div
                        class="other_instructor_view-img"
                      >
                        <img
                          alt=""
                          src="profile-img.jpg"
                        />
                      </div>
                      <div
                        class="other_instructor_view-name-career"
                      >
                        <h2>
                          Joginder Signh
                        </h2>
                        <span>
                          Web Developer, Designer, and Teacher
                        </span>
                      </div>
                    </div>
                    <ul
                      class="other_instructor_view-more_info"
                    >
                      <li>
                        <div
                          class="other_instructor_view-enroll_label_more_info"
                        >
                          <div
                            class="other_instructor_view-label"
                          >
                            Enroll Students
                          </div>
                          <div
                            class="other_instructor_view-number"
                          >
                            612K
                          </div>
                        </div>
                      </li>
                      <li>
                        <div
                          class="other_instructor_view-enroll_label_more_info"
                        >
                          <div
                            class="other_instructor_view-label"
                          >
                            Courses
                          </div>
                          <div
                            class="other_instructor_view-number"
                          >
                            8
                          </div>
                        </div>
                      </li>
                      <li>
                        <div
                          class="other_instructor_view-enroll_label_more_info"
                        >
                          <div
                            class="other_instructor_view-label"
                          >
                            Reviews
                          </div>
                          <div
                            class="other_instructor_view-number"
                          >
                            115K
                          </div>
                        </div>
                      </li>
                      <li>
                        <div
                          class="other_instructor_view-enroll_label_more_info"
                        >
                          <div
                            class="other_instructor_view-label"
                          >
                            Subscribers
                          </div>
                          <div
                            class="other_instructor_view-number"
                          >
                            452K
                          </div>
                        </div>
                      </li>
                    </ul>
                  </div>
                  <div
                    class="other_instructor_view-col-lg-5"
                  >
                    <a
                      class="other_instructor_view-report_profile"
                      href="#"
                    >
                      <span>
                        <i
                          class="uil uil-windsock"
                        />
                      </span>
                      Setting
                    </a>
                    <div
                      class="other_instructor_view-social"
                    >
                      <ul
                        class="social-icons"
                      >
                        <a
                          class="oiv-fb"
                          href="#"
                        >
                          <img
                            src="facebook.png"
                          />
                        </a>
                        <a
                          class="oiv-x"
                          href="#"
                        >
                          <img
                            src="x.png"
                          />
                        </a>
                        <a
                          class="oiv-ln"
                          href="#"
                        >
                          <img
                            src="linkedin.png"
                          />
                        </a>
                        <a
                          class="oiv-yt"
                          href="#"
                        >
                          <img
                            src="youtube.png"
                          />
                        </a>
                      </ul>
                    </div>
                    <ul
                      class="other_instructor_view-subcrise"
                    >
                      <button
                        class="subscribe-btn btn500"
                      >
                        Cursus Studio
                      </button>
                      <button
                        class="msg125 btn500"
                      >
                        Edit
                      </button>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="other_instructor_view-navigation"
      >
        <div
          class="other_instructor_view-container"
        >
          <div
            class="other_instructor_view-row"
          >
            <div
              class="other_instructor_view-col-lg-12"
            >
              <div
                class="other_instructor_view-course_tabs"
              >
                <nav>
                  <div
                    class="other_instructor_view-nav"
                  >
                    <div
                      class="tab active"
                    >
                      <div
                        class="oiv-nav-about"
                      >
                        About
                      </div>
                    </div>
                    <div
                      class="tab "
                    >
                      <div
                        class="oiv-nav-courses"
                      >
                        Courses
                      </div>
                    </div>
                    <div
                      class="tab "
                    >
                      <div
                        class="oiv-nav-courses"
                      >
                        Purchased
                      </div>
                    </div>
                    <div
                      class="tab "
                    >
                      <div
                        class="oiv-nav-discussion"
                      >
                        Discussion
                      </div>
                    </div>
                    <div
                      class="tab "
                    >
                      <div
                        class="oiv-nav-courses"
                      >
                        Subscriptions
                      </div>
                    </div>
                  </div>
                  <div
                    class="about-tab"
                  >
                    <div
                      class="tab-pane fade show active"
                    >
                      <div
                        class="other_instructor_view-about_me"
                      >
                        <div
                          class="label-about"
                        >
                          <h3>
                            About Me
                          </h3>
                          <p>
                            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum scelerisque nibh sed ligula blandit, quis faucibus lorem pellentesque. Suspendisse pulvinar dictum pellentesque. Vestibulum at sagittis lectus, sit amet aliquam turpis. In quis elit tempus, semper justo vitae, lacinia massa. Etiam sagittis quam quis fermentum lacinia. Curabitur blandit sapien et risus congue viverra. Mauris auctor risus sit amet cursus sollicitudin. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla feugiat sodales massa, in viverra dolor condimentum ut. In imperdiet, justo nec volutpat blandit, tellus justo tempor quam, sed pretium nibh nunc nec mauris. Mauris vel malesuada magna. Quisque iaculis molestie purus, non luctus mauris porta id. Maecenas imperdiet tincidunt mauris vestibulum vulputate. Aenean sollicitudin pretium nibh, et sagittis risus tincidunt ac. Phasellus scelerisque rhoncus massa, ac euismod massa pharetra non. Phasellus dignissim, urna in iaculis varius, turpis libero mollis velit, sit amet euismod arcu mi ac nibh. Praesent tincidunt eros at ligula pellentesque elementum. Fusce condimentum enim a tellus egestas, sit amet rutrum elit gravida. Pellentesque in porta sapien. Fusce tristique maximus ipsum et mollis. Sed at massa ac est dapibus vulputate at eu nibh.
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </nav>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
